### PR TITLE
= 1.7.0 =

### PR DESCRIPTION
**Pull request by arniebradfo**
Pulled the newest dev branch off the WordPress subversion repo and merged it with this. Please merge it in here so folks can fork your code.

### CODE REVIEW

**Comment by mukhortov** [comment]
Thanks for your contribution!

Before I can merge it would be great to fix the following issues in this PR:

1. There is a grey bar in the middle of the editor when `Enable full-height editor and distraction-free functionality.` option is enabled:
screen shot 2016-05-26 at 20 40 56

2. Editor theme unexpectedly changes when you change the font size. To reproduce it change the theme, then change font size. 

3. Is there a reason to change Less to Sass? It was used just to concatenate files, but anyway, I'd prefer to keep Less.

**Comment by arniebradfo** [comment]
Ok Peter,

I fixed those bugs. The bug with the font changing the theme was caused by some code with the comment `// Dirty Fix`(see `dev/hesh.dev.js` line 279). Do you remember what that was a fix for? Everything seems to run normally without out it.

Your also going to have to check the bug with the gray bar. I couldn't recreate it, but i think i fixed it.

Check it and let me know if we're good. 

Thanks,
-James

File: dev/hesh.dev.js [hunk 1]
```diff
@@ -277,11 +276,10 @@
...
 				changeFontSize();
-				// Dirty Fix
-				editor.toTextArea();
-				runEditor();
-				editor.focus();
-
+				// Dirty Fix // fix for what?
+				// editor.toTextArea();
+				// runEditor();
+				// editor.focus();
```
Comment (mukhortov) [inline dev/hesh.dev.js]: Without it, cursor positioning goes mad after the font size is changed. Try to set the font size to 22, reload the page then change the font to a smaller one, eg 12. And click on different lines.

Back then I haven't found a better solution to fix that without reloading the Codemirror plugin. It's quite a silly way of fixing it, that's why it's "Dirty fix" :) Maybe you can find a better solution.

File: dev/hesh.dev.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+			var selector = document.getElementById('cm_font_size');
+			changeFontSize();
+			selector.value = fontSize;
+			selector.onchange = function () {
+				fontSize = this.value;
+				changeFontSize();
+				// Dirty Fix // fix for what?
+				// editor.toTextArea();
+				// runEditor();
+				// editor.focus();
```
Comment (mukhortov) [inline dev/hesh.dev.js]: Without it, cursor positioning goes mad after the font size is changed. Try to set the font size to 22, reload the page then change the font to a smaller one, eg 12. And click on different lines.

Back then I haven't found a better solution to fix that without reloading the Codemirror plugin. It's quite a silly way of fixing it, that's why it's "Dirty fix" :) Maybe you can find a better solution.

**Comment by mukhortov** [comment]
> Do you remember what that was a fix for? Everything seems to run normally without out it.

it's for fixing cursor positioning after the font size is changed. There is more detailed explanation in the comment above

> Your also going to have to check the bug with the gray bar. I couldn't recreate it, but I think I fixed it.

Yes, it's fixed!

**Comment by arniebradfo** [comment]
Oh that was a weird bug. 

Fixed now.

Merge time.

**Comment by mukhortov** [comment]
Awesome! Going to merge and release! 👍

**Pull request merged**